cumulusci/tasks/bulkdata/snowfakery.py

```diff
@@ -0,0 +1,657 @@
+                f"Errors exceeded threshold: {upload_status.sets_failed} vs {ERROR_THRESHOLD}"
+            )
+
+        if (
+            org_record_counts_thread
+            and org_record_counts_thread.other_inaccurate_record_counts
+        ):
+            for k, v in org_record_counts_thread.other_inaccurate_record_counts.items():
+                self.logger.info(f"      COUNT: {k}: {v:,}")
+
+        return upload_status
+
+    def tick(
+        self,
+        upload_status,
+        template_path,
+        tempdir,
+        portions: PortionGenerator,
+    ):
+        """Called every few seconds, to make new data generators if needed."""
+        self.data_gen_q.tick()
+
+        if portions.done(upload_status.total_sets_working_on_or_uploaded):
+            self.logger.info("Finished Generating")
+        # queue is full
+        elif self.data_gen_q.full:
+            self.logger.info("Waiting before datagen (queue full)")
+        else:
+            for i in range(self.data_gen_q.num_free_workers):
+                upload_status = self.generate_upload_status(
+                    portions.next_batch_size,
+                    template_path,
+                )
+                self.job_counter += 1
+                batch_size = portions.next_batch(
+                    upload_status.total_sets_working_on_or_uploaded
+                )
+                if not batch_size:
+                    self.logger.info(
+                        "All scheduled portions generated and being uploaded"
+                    )
+                    break
+                job_dir = self.generator_data_dir(
+                    self.job_counter, template_path, batch_size, tempdir
+                )
+                self.data_gen_q.push(job_dir)
+
+    def finish(self, upload_status, data_gen_q, load_data_q):
+        """Wait for jobs to finish"""
+        while data_gen_q.workers + load_data_q.workers:
+            self.logger.info(
+                f"Waiting for {len(data_gen_q.workers)} datagen, {len(load_data_q.workers)} upload workers to finish"
+            )
+            data_gen_q.tick()
+            time.sleep(WAIT_TIME)
+
+        self.log_failures(set(load_data_q.failed_job_dirs + data_gen_q.failed_job_dirs))
+        elapsed = format_duration(timedelta(seconds=time.time() - self.start_time))
+
+        if self.run_until.sobject_name:
+            set_name = f"{self.run_until.sobject_name} records and associated records"
+        else:
+            set_name = "iterations"
+
+        self.logger.info(
+            f"☃ Snowfakery created {elapsed}, {upload_status.target_count:,} {set_name}"
+        )
+
+    def log_failures(self, dirs=T.Sequence[Path]):
+        """Log failures from sub-processes to main process"""
+        for failure_dir in dirs:
+            f = Path(failure_dir) / "exception.txt"
+            if not f.exists():
+                continue
+            exception = f.read_text().strip().split("\n")[-1]
```

Comment:
```suggestion
            exception = f.read_text(encoding="utf-8").strip().split("\n")[-1]
```

Reply:
There are also some calls to `open()` which need the same fix.

Reply:
f31f2e2354330e4b338b1282153028683576b41a